src/core/edm/utils/useEntityTypePropertyTypes.js

```diff
@@ -0,0 +1,46 @@
+/*
+ * @flow
+ */
+
+import { Map } from 'immutable';
+import { Models } from 'lattice';
+import { ValidationUtils } from 'lattice-utils';
+import { useSelector } from 'react-redux';
+import type { UUID } from 'lattice';
+
+import {
+  EDM,
+  ENTITY_TYPES,
+  ENTITY_TYPES_INDEX_MAP,
+  PROPERTY_TYPES,
+  PROPERTY_TYPES_INDEX_MAP,
+} from '../../redux/constants';
+
+const { EntityType, FQN, PropertyType } = Models;
+const { isValidUUID } = ValidationUtils;
+
```

Comment:
This would be a great candidate to try using https://github.com/reduxjs/reselect

Reply:
https://github.com/openlattice/lattice-orgs/pull/1084#discussion_r654158157